AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTests.m

```diff
@@ -329,6 +365,45 @@ - (void)testTrackEventWithoutProperties {
   assertThat(name, is(expectedName));
 }
 
+- (void)testTrackEventWhenAnalyticsDisabled {
+
+  // If
+  id analyticsMock = OCMPartialMock([MSAnalytics sharedInstance]);
+  id logManagerMock = OCMProtocolMock(@protocol(MSLogManager));
+  OCMStub([analyticsMock isEnabled]).andReturn(NO);
+  [MSAppCenter configureWithAppSecret:kMSTestAppSecret];
+  [[MSAnalytics sharedInstance] startWithLogManager:logManagerMock appSecret:kMSTestAppSecret];
+
+  // FIXME: logManager holds session tracker somehow and it causes other test failures. Stop it for hack.
+  [[MSAnalytics sharedInstance].sessionTracker stop];
+
+  // When
+  OCMReject([logManagerMock processLog:OCMOCK_ANY forGroupId:OCMOCK_ANY]);
+  [[MSAnalytics sharedInstance] trackEvent:@"Some event" withProperties:nil];
+
+  // Then
+  OCMVerifyAll(logManagerMock);
+}
+
+- (void)testTrackEventWithInvalidName {
+
+  // If
+  NSString *invalidEventName = nil;
+  id logManagerMock = OCMProtocolMock(@protocol(MSLogManager));
+  [MSAppCenter configureWithAppSecret:kMSTestAppSecret];
+  [[MSAnalytics sharedInstance] startWithLogManager:logManagerMock appSecret:kMSTestAppSecret];
+
+  // FIXME: logManager holds session tracker somehow and it causes other test failures. Stop it for hack.
+  [[MSAnalytics sharedInstance].sessionTracker stop];
+
+  // When
+  OCMReject([logManagerMock processLog:OCMOCK_ANY forGroupId:OCMOCK_ANY]);
+  [[MSAnalytics sharedInstance] trackEvent:invalidEventName withProperties:nil];
```

Comment:
Add more verifies here.
`validateEventName:forLogType:`, `validateProperties:forLogName:andType:` should be called in this test.

Reply:
Same for trackPage test